### PR TITLE
Enable alternative author and date via cli parameters

### PR DESCRIPTION
**Pull request by neithernut**
git commit` allows setting the author and date of a commit via command line parameters. This PR introduces a similar interface for the "create-message", "new" and "reply" subcommands. The new interface may be used for scripting, e.g. for implementing a mail-to-message bridge.

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -22,6 +22,11 @@ error_chain! {
             display("Malformed filter spec: {}", spec)
         }
 
+        MalformedAuthor(author: String) {
+            description("Malformed author")
```
Comment (matthiasbeyer) [inline src/error.rs]: We should maybe state how the format should be.

File: src/util.rs [hunk 2]
```diff
@@ -179,6 +184,55 @@ impl<'r> RepositoryUtil<'r> for Repository {
...
 
+    fn cli_author(&self, matches: &ArgMatches) -> git2::Signature {
+        use chrono::{DateTime, FixedOffset};
+
+        // First, get a signature appropriate for the author
+        let author = matches
+            .value_of("author")
+            .map(|a| {
+                // NOTE: this function will be called only once per run anyway
+                let parts = Regex::new(r"^([^<>]+)[[:space:]]+<([^[:space:]@]+@[^[:space:]@]+)>$")
```
Comment (matthiasbeyer) [inline src/util.rs]: This regex could be put either in the outer scope or even as constant into a `lazy_static`, so the regex is only compiled once per call of `cli_author()` or even only once per `git-dit` execution!

Comment (neithernut) [reply src/util.rs]: Have you even read the note one line above? If this function is called more than once per run, the dev using it is doing something wrong.

**Comment by neithernut** [comment]
I think that's purely a fault of the documentation. The code does override the author date, not the committer date.

File: src/util.rs (hunk 2 above)
Comment (matthiasbeyer) [reply src/util.rs]: Whops, jup.

**Pull request merged**